friends/src/App.js

```diff
@@ -0,0 +1,98 @@
+import React, { Component } from 'react';
+import './App.css';
+import axios from 'axios';
+
+import FriendsList from './FriendsList';
+import FriendForm from './FriendForm'
+
+class App extends Component {
+  constructor(props){
+    super(props);
+    this.state = {
+        friends: [],
+        
+    };
+  }
+  
+  componentDidMount() {
+    axios
+        .get('http://localhost:5000/friends')
+        .then(response => {
+            this.setState(() => ({ friends: response.data }));
+        })
+        .catch(error => {
+          console.error('Ummm...what...!?!', error);
+        });
+  }
+  
+  inputHandler = e => {
+    this.setState({ [e.target.name]: e.target.value })
+  };
+  
+  addFriend = event => {
+    const { name, age, email } = this.state;
+    event.preventDefault();
+    axios.post('http://localhost:5000/friends', {name, age, email})
+    .then(response => {
+      this.setState({ friends: response.data, name: "", age: "", email: "" });
+      window.location.reload();
+    })
+    .catch(err => console.log("Umm...that's a problem", err));
+  };
+  
+  deleteFriend = (id) => {
+    return () => {
+      axios.delete(`http://localhost:5000/friends/${id}`)
+      .then(response => {
+        this.setState({friends: response.data})
+      })
+      .catch(err => console.log("You can't get rid of me that easily!"))
+    }
+
+  }
+  
+  updateHandler = event => {
+    const { name, age, email } = this.state;
+    const id = this.props.friend.id
+    event.preventDefault();
```

Comment:
This function looks good but it needs to be in the FriendsDetails component in order to function properly. You are trying to access the specific ID of the element whose update button you click.